admin-dev/themes/new-theme/js/pages/currency/form/currency-form.js

```diff
@@ -0,0 +1,133 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import { showGrowl } from '../../../app/utils/growl';
+
+export default class CurrencyForm {
+  /**
+   * @param {object} currencyFormMap - Page map
+   */
+  constructor(currencyFormMap) {
+    this.map = currencyFormMap;
+    this.$currencyForm = $(this.map.currencyForm);
+    this.getCLDRDataUrl = this.$currencyForm.data('get-cldr-data');
+    this.$currencySelector = $(this.map.currencySelector);
+    this.$isUnofficialCheckbox = $(this.map.isUnofficialCheckbox);
+    this.$isoCodeInput = $(this.map.isoCodeInput);
+    this.$exchangeRateInput = $(this.map.exchangeRateInput);
+    this.$precisionInput = $(this.map.precisionInput);
+    this.$resetDefaultSettingsButton = $(this.map.resetDefaultSettingsInput);
+    this.$loadingDataModal = $(this.map.loadingDataModal);
+  }
+
+  init() {
+    this._initListeners();
+    this._initFields();
+  }
+
+  _initListeners() {
+    this.$currencySelector.change(this._onCurrencySelectorChange.bind(this));
+    this.$isUnofficialCheckbox.change(this._onIsUnofficialCheckboxChange.bind(this));
+    this.$resetDefaultSettingsButton.click(this._onResetDefaultSettingsClick.bind(this));
+  }
+
+  _initFields() {
+    if (!this._isUnofficialCurrency()) {
+      this.$isUnofficialCheckbox.prop('checked', false);
+      this.$isoCodeInput.prop('readonly', true);
+    } else {
+      this.$currencySelector.val('');
+      this.$isoCodeInput.prop('readonly', false);
+    }
+  }
+
+  _onCurrencySelectorChange() {
+    const selectedISOCode = this.$currencySelector.val();
+    if ('' !== selectedISOCode) {
+      this.$isUnofficialCheckbox.prop('checked', false);
+      this.$isoCodeInput.prop('readonly', true);
+      this._resetCurrencyData(selectedISOCode);
+    } else {
+      this.$isUnofficialCheckbox.prop('checked', true);
+      this.$isoCodeInput.prop('readonly', false);
+    }
+  }
+
+  _isUnofficialCurrency() {
+    if ('hidden' === this.$isUnofficialCheckbox.prop('type')) {
+      return 0 !== parseInt(this.$isUnofficialCheckbox.attr('value'));
+    }
+
+    return this.$isUnofficialCheckbox.prop('checked');
+  }
+
+  _onIsUnofficialCheckboxChange() {
+    if (this._isUnofficialCurrency()) {
+      this.$currencySelector.val('');
+      this.$isoCodeInput.prop('readonly', false);
+    } else {
+      this.$isoCodeInput.prop('readonly', true);
+    }
+  }
+
+  _onResetDefaultSettingsClick() {
+    this._resetCurrencyData(this.$isoCodeInput.val());
+
+    return false;
```

Comment:
What is the reason of returning false here?

Reply:
Because it was previously a button, so I need to prevent the form submit, now that it's an `a` I can remove it

Reply:
use preventDefault instead :thinking: